addons/survey/static/src/js/survey_session_chart.js

```diff
@@ -229,8 +229,13 @@ publicWidget.registry.SurveySessionChart = publicWidget.Widget.extend({
                         let resultLines = [];
                         let currentLine = [];
                         for (let i = 0; i < words.length; i++) {
-                            // If the word we are adding exceed already the number of characters for the line, we add it anyway before passing to a new line
-                            currentLine.push(words[i]);
+                            // Chop down words that do not fit on a single line, add each part on its own line.
```

Comment:
Quick question, as you now split big words into smaller chunks, does it have an impact on the logic here above, that is using the longest word of the text to compute the fontRatio?

Reply:
The computation "Adapt font size if..." works as follow: if the longest word doesn't fit on a single line then the ratio is changed to a hardcoded default of `0.4`, the fontSize doesn't *scale* with the length of the longest word.

That hardcoded value: 0.4, is already the default fontRatio when there are 1->4 columns so it really only impactes the rendering when there are 5 columns and more. But starting at 5 columns, only 1 of every 2 labels is shown on screen, regardless of this task.

So we are picking a smaller font when there are too-long words and at the same time we only show half of the labels which actually leave rooms for a bigger font. IMO that "Adapt font size if..." block could be removed now that we just chop down those long words.

Reply:
Can you take a look then and remove it if no longer necessary?
If we can simplify this a bit then I'm all for it.

Wouldn't we still need to reduce the font size if there are a lot of words though, even if we chop them down, to avoid having the label taking a lot of height in the end? (Chopping down words can make this 'worse' now that I think about it).
Not sure this was part of the previous implementation actually, but maybe it happened less often as we had most likely some long words in there, meaning the font ratio was reduced?

(Don't forget to adapt the docstring if so).

Reply:
Well it looks like reducing the font size *is* a good idea

![2023-11-14_11-58](https://github.com/odoo/odoo/assets/8208953/bc7dd234-a6cd-4bf1-b152-4462724a0d2d)
![2023-11-14_11-59](https://github.com/odoo/odoo/assets/8208953/555643d6-6bf7-40ea-8e9a-bc85dc4a193d)

Reply:
😄 

So we keep the current code version then, best of both worlds?

(Love the sample data by the way).

Reply:
Yes let's do that :)